src/core/featuremodel.cpp

```diff
@@ -31,12 +31,71 @@ FeatureModel::FeatureModel( QObject *parent )
   connect( this, &FeatureModel::modelReset, this, &FeatureModel::featureChanged );
 }
 
+void FeatureModel::setModelMode( const ModelModes mode )
+{
+  if ( mModelMode == mode )
+    return;
+
+  beginResetModel();
+  mModelMode = mode;
+  emit modelModeChanged();
+  endResetModel();
+}
+
+FeatureModel::ModelModes FeatureModel::modelMode() const
+{
+  return mModelMode;
+}
+
 void FeatureModel::setFeature( const QgsFeature &feature )
 {
+  if ( mModelMode != SingleFeatureModel || mFeature == feature )
+    return;
+
   beginResetModel();
   mFeature = feature;
-  endResetModel();
+  mFeatures.clear();
+  mAttributesAllowEdit.clear();
   emit featureChanged();
+  endResetModel();
+}
+
+void FeatureModel::setFeatures( const QList<QgsFeature> &features )
+{
+  if ( mModelMode != MultiFeatureModel )
+    return;
+
+  beginResetModel();
+  if ( !features.isEmpty() )
+  {
+    mFeatures = features;
+    mAttributesAllowEdit.clear();
+
+    mFeature = mFeatures.at( 0 );
+    for ( int i = 0; i < mFeature.attributes().count(); i++ )
+    {
+      bool equalValue = true;
+      for ( const QgsFeature &feature : mFeatures )
```

Comment:
```suggestion
      if ( std::any_of( mFeatures.begin(), mFeatures.end(), [this, i]( const QgsFeature &feature ) { return mFeature.attributes().at( i ) != feature.attributes().at( i ); } )
      {
        mFeature.setAttribute( i, QVariant() );
        equalValue = false;
      }
```

I think that's what cppcheck asks us to do. (Written without a compiler)